src/main/java/org/purescript/psi/PSModule.kt

```diff
@@ -87,6 +86,32 @@ class PSModule(node: ASTNode) :
             return exportedValueDeclarations
         }
 
+    /**
+     * All the foreign value declarations that this module exports,
+     * both directly and through re-exported modules
+     */
+    val exportedForeignValueDeclarations: List<PSForeignValueDeclaration>
+        get() {
+            val explicitlyExportedItems = exportList?.exportedItems
+                ?: return foreignValueDeclarations.toList()
+
+            val explicitlyExportedValueNames = explicitlyExportedItems
+                .filterIsInstance<PSExportedValue>()
+                .map { it.name }
+                .toSet()
+
+            val exportedForeignValueDeclarations = mutableListOf<PSForeignValueDeclaration>()
+            foreignValueDeclarations.filterTo(exportedForeignValueDeclarations) {
+                it.name in explicitlyExportedValueNames
+            }
+
+            explicitlyExportedItems.filterIsInstance<PSExportedModule>()
+                .mapNotNull { it.importDeclaration }
+                .flatMapTo(exportedForeignValueDeclarations) { it.importedForeignValueDeclarations }
+
+            return exportedForeignValueDeclarations
+        }
+
```

Comment:
This property is almost identical to `exportedValueDeclarations`. We could look for some fancy abstraction, but for now I think it's alright.